tests/test_plain_datetime.py

```diff
@@ -64,6 +64,66 @@ def test_iso(self):
             2020, 8, 15, 5, 12, 30, nanosecond=450
         )
 
+    def test_leap_seconds_parsing(self):
+        # Leap second (60) should be parsed and normalized to 59
+        assert PlainDateTime("2020-08-15T05:12:60") == PlainDateTime(
+            2020, 8, 15, 5, 12, 59
+        )
+        assert PlainDateTime("2020-08-15T05:12:60.123456") == PlainDateTime(
+            2020, 8, 15, 5, 12, 59, nanosecond=123_456_000
+        )
+        # Basic format
+        assert PlainDateTime("20200815T051260") == PlainDateTime(
+            2020, 8, 15, 5, 12, 59
+        )
+        # Direct construction should still reject 60
+        with pytest.raises(ValueError):
+            PlainDateTime(2020, 8, 15, 5, 12, 60)
+
+    def test_leap_seconds_comprehensive(self):
+        # Extended format with various fractional seconds
+        assert PlainDateTime.parse_iso(
+            "2020-08-15T23:59:60.999999999"
+        ) == PlainDateTime(2020, 8, 15, 23, 59, 59, nanosecond=999_999_999)
+        assert PlainDateTime.parse_iso(
+            "2020-08-15T12:34:60.5"
+        ) == PlainDateTime(2020, 8, 15, 12, 34, 59, nanosecond=500_000_000)
+
+        # Basic format with fractional seconds
+        assert PlainDateTime.parse_iso(
+            "20200815T123460.123456"
+        ) == PlainDateTime(2020, 8, 15, 12, 34, 59, nanosecond=123_456_000)
+
+        # Comma as decimal separator
+        assert PlainDateTime.parse_iso(
+            "2020-08-15T12:34:60,5"
+        ) == PlainDateTime(2020, 8, 15, 12, 34, 59, nanosecond=500_000_000)
+
+        # Various dates with leap seconds
+        assert PlainDateTime.parse_iso("1999-12-31T23:59:60") == PlainDateTime(
+            1999, 12, 31, 23, 59, 59
+        )
+        assert PlainDateTime.parse_iso("2012-06-30T23:59:60") == PlainDateTime(
+            2012, 6, 30, 23, 59, 59
+        )
+        assert PlainDateTime.parse_iso("2015-06-30T23:59:60") == PlainDateTime(
+            2015, 6, 30, 23, 59, 59
+        )
+
+    def test_leap_seconds_invalid_datetimes(self):
+        # 61 and above should still be rejected
+        with pytest.raises(ValueError, match="Invalid format"):
+            PlainDateTime.parse_iso("2020-08-15T12:34:61")
+        with pytest.raises(ValueError, match="Invalid format"):
+            PlainDateTime.parse_iso("2020-08-15T12:34:99")
+
+    def test_leap_seconds_normal_seconds_still_work(self):
+        # Ensure normal seconds 00-59 still parse correctly
+        for sec in range(60):
```

Comment:
this test is a bit overkill. You've already defined a good spread of cases above